deep_qa/data/instances/text_classification/frame_instance.py

```diff
@@ -0,0 +1,198 @@
+from typing import Dict, List
+
+import numpy
+from overrides import overrides
+
+from ..instance import TextInstance, IndexedInstance
+from ...data_indexer import DataIndexer
+
+# TODO PR request for having these in the json as an application specific content
+# the slotnames can vary according to different end applications, e.g., a HowTo tuple, OpenIE tuple ...
+SLOTNAMES_ORDERED = ["agent", "beneficiary", "causer", "context", "definition", "event",
```

Comment:
Oh, yeah, your preprocess scripts putting everything into this format will take care of that, I think.

Another consideration is that you'll be essentially wasting computation and modeling capacity on all of the un-filled slots by filling them with "unk".  You could either: (1) leave them empty, and let them be masked in whatever model computation you do later.  This will keep the model from wasting its capacity trying to figure out that "unk" means "unfilled".  Or (2) instead of having a fixed-sized input here, you could have a key-value input, where you give a list of (slot, value) pairs as input to the model.  In this setting, it's also a lot easier to think about how to pass in the query - it's just another (slot, value) pair, just with a `"QUERY"` value, or something similar.  Then instead of learning what is essentially an embedding of each slot type in the weights of a softmax layer, you can explicitly learn an embedding for them, and have that be part of your model.  Happy to talk about this more in person if it's hard to follow what I'm suggesting here.